salt/config/__init__.py

```diff
@@ -2852,7 +2852,7 @@ def is_profile_configured(opts, provider, profile_name, vm_=None):
         non_size_drivers.append('linode')
 
     # If cloning on VMware, specifying image is not necessary.
-    if driver == 'vmware' and profile_key.get('image', True):
+    if driver == 'vmware' and 'image' not in list(profile_key.keys()) and 'clonefrom' in list(profile_key.keys()):
```

Comment:
I see the point about `non_image_drivers`, thanks for the suggestion for this line, it was pretty close.

I have to admit I'm finding that the double negatives (i.e. `driver not in non_size_drivers`) are pretty confusing, though I understand it's because there are fewer cases where `image` is not required. 

That said, I was wondering if there were plans afoot to hand this profile checking off to the individual drivers rather than embedding per-driver logic outside of the drivers themselves?